Makefile

```diff
@@ -60,13 +63,13 @@ sphinx:
 	make -f docs/Makefile.sphinx html
 
 lint:
-	ruff check c7n tests tools
-	black --check $(FMT_SET)
+	uv run --no-project ruff check c7n tests tools
```

Comment:
how about [uvx ruff](https://docs.astral.sh/uv/guides/tools/#running-tools)?

Reply:
we did uvx for bandit and semgrep analyzers, for ruff its a little less clear to me as its a minimal dependency set, I'm unclear how uvx works with cache in ci vs standard uv sync / uv run. conceptually it does make sense, as does uv tool, although I'm a little less clear on the distinction between the those two. still learning uv.

Reply:
Treating ruff as a tool run would let us bump versions without impacting uv.lock. That might be useful?

Aside from that, agreed that ruff _feels_ like a tool run even though the practical impact of the switch seems minimal. So like the explicit `cache-suffix` lines above, this feels like a case where either way works.